src/OpenApi/Generator/Client/ServerPluginGenerator.php

```diff
@@ -33,6 +33,14 @@ private function discoverServer(OpenApi $openApi): array
                 $plugins[] = AddHostPlugin::class;
             }
 
+            $variables = $server->getVariables();
+
+            if (null !== $variables && \array_key_exists('port', $variables)) {
+                if (null !== $variables['port']->getDefault()) {
```

Comment:
This condition could be merged in the if before

Reply:
Sure, I just change it !